lib/livebook_web/live/path_select_component.ex

```diff
@@ -27,7 +30,7 @@ defmodule LivebookWeb.PathSelectComponent do
     dir = Path.expand(dir)
 
     files =
-      if assigns.current_dir != dir do
+      if assigns.current_dir != dir or assigns[:force_reload] do
```

Comment:
Unfortunately we are storing force_reload in the assigns forever, so it will always be force reloaded. My suggestion is to add a line at the top of `update/2` like this:

    {force_reload?, assigns} = Map.pop(assigns, :force_reload, false)